codegen/src/main/java/software/amazon/awssdk/codegen/naming/DefaultNamingStrategy.java

```diff
@@ -302,6 +302,11 @@ public String getEnumValueName(String enumValue) {
         // Special cases
         result = result.replace("textORcsv", "TEXT_OR_CSV");
 
+        // leading digits, add a prefix
+        if (result.matches("^\\d.*")) {
+            result = "VALUE_" + result;
```

Comment:
I understand the java limitation of not being able to have a number as an enum value, but can you explain how this substitution works in practice?